### PR TITLE
Don't have RPC clients block on connect

### PR DESCRIPTION
**Pull request by lerks**
This is my version of the fix proposed in #814. I had shared a simpler version on Gitter but some more changes were required in order to make the tests pass.

---
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/1014)

### CODE REVIEW

**Comment by stefano-maggiolo** [comment]
LGTM, shall we ask @edomora97 and @tuankiet65 if this fixes their issues?

**Comment by tuankiet65** [comment]
Thanks for having me, I'll try to test this out by next week (we're having a contest this Sunday so I don't feel like trying it out right now). Also can I just patch this directly on top of v1.3.2 or I have to use the master branch?

**Comment by lerks** [comment]
@tuankiet65 Thanks for checking!

@stefano-maggiolo I realized that the unit test wasn't really testing the right thing, so I fixed it. I also "merged" `connected` and `connected_event`. PTAL

File: cmstestsuite/unit_tests/io/rpc_test.py [hunk 1]
```diff
@@ -239,42 +239,56 @@ def test_method_return_list(self):
         self.assertTrue(result.successful())
         self.assertEqual(result.value, ["Hello", 42, "World"])
 
-    def test_background_connect(self):
-        self.kill_listener()
-        client = self.get_client(ServiceCoord("Foo", 0), block=False)
-        self.assertFalse(client.connected)
-        self.assertFalse(client.connected_event.is_set())
+    @patch("cms.io.rpc.gevent.socket.socket")
+    def test_background_connect(self, socket_mock):
```
Comment (stefano-maggiolo) [inline cmstestsuite/unit_tests/io/rpc_test.py]: Comment a bit what it's doing.

File: cms/io/rpc.py [hunk 2]
```diff
@@ -94,8 +94,7 @@ def __init__(self, remote_address):
 
         """
         self.remote_address = remote_address
-        self.connected = False
-        self.connected_event = gevent.event.Event()
+        self.connected = gevent.event.Event()
```
Comment (stefano-maggiolo) [inline cms/io/rpc.py]: This was a somehow public-ish api (hence all the places you had to change). Maybe a property connected that looks at _connected_event.is_set()?

**Comment by stefano-maggiolo** [comment]
Nit: there are a few places in the test that you changed than reverted, and eventually only the formatting changed, if you want to revert them fully, feel free, otherwise I don't care that much.

**Pull request merged**